Execution/src/ExecutionController.js

```diff
@@ -44,21 +48,44 @@ const executePython = async (req, res) => {
     fs.writeFileSync(scriptPath, pythonCode); // Write the code to a temporary python file
 
     const pythonProcess = spawn('python', [scriptPath]); // Command to execute the script
-    const result = await executeScript(scriptPath, pythonProcess);
+    const result = await executeScript(scriptPath, pythonProcess, false);
+
+    logger.logSuccess("Program executed successfully.");
 
     res.json({ output: result });
   } catch (err) {
     logger.log(err);
   }
 };
 
-// execute java
 const executeJava = async (req, res) => {
   try {
-    console.log(req.body);
-    res.json({ message: 'java code executed' });
+    const javaCode = req.body.code;
+    const javaFilename = 'Main.java';
+    const javaClassName = 'Main';
+
+    // Write the Java source code to a .java file
+    fs.writeFileSync(javaFilename, javaCode);
+
+    // Compile the Java source code
+    const compileProcess = spawn('javac', [javaFilename]);
+    const compileResult = await executeScript(javaFilename, compileProcess, false);
+
+    if (compileResult.includes('error')) {
+      logger.error('Compilation failed.');
+      res.json({ output: compileResult });
+    } else {
+      logger.logSuccess("Program compiled successfully.");
+
+      const javaProcess = spawn('java', [javaClassName]);
+      const result = await executeScript(javaClassName, javaProcess, true);
```

Comment:
(2/2) ... perhaps you can modify it here first, before calling `executeScript`.

i.e. do something like `const result= await executeScript(javaFilename + '.class', compileProcess);`

Reply:
updated